subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -287,4 +288,17 @@ public void applyTo(boolean value, StartParameter settings, Origin origin) {
             }
         }
     }
+
+    public static class DependencyLockOption extends BooleanBuildOption<StartParameter> {
+        public static final String LONG_OPTION = "dependency-lock";
+
+        public DependencyLockOption() {
+            super(null, BooleanCommandLineOptionConfiguration.create(LONG_OPTION, "Turns on dependency locking. Gradle will store locks for resolved dependencies and reuse them in subsequent builds.", "Disables storing and using locks for dependencies.").incubating());
```

Comment:
❌ The option should only be about generating locks. The lock should automatically be used if it is present.

Reply:
done